src/midiEditorCommands.cpp

```diff
@@ -1060,17 +1061,38 @@ void cmdMidiNoteSplitOrJoin(Command* command) {
 }
 
 #ifdef _WIN32
+map<string, string> parseEventData(string const& source) {
+	map<string, string> m;
+	string key, val;
+	istringstream s(source);
+	while(getline(getline(s, key, '='), val, ' ')) {
+		m[key] = val;
+	}
+	return m;
+}
+
 void cmdFocusNearestMidiEvent(Command* command) {
 	HWND focus = GetFocus();
 	if (!focus)
 		return;
 	double cursorPos = GetCursorPosition();
-	for (int i = 0; i < ListView_GetItemCount(focus); ++i) {
-		char text[50];
-		// Get the text from the position column (1).
-		ListView_GetItemText(focus, i, 1, text, sizeof(text));
-		// Convert this to project time. text is always in measures.beats.
-		double eventPos = parse_timestr_pos(text, 2);
+	HWND editor = MIDIEditor_GetActive();
+	assert(editor == GetParent(focus));
+	for (int i = 0; i < MIDIEditor_GetSetting_int(editor, "list_cnt"); ++i) {
```

Comment:
Please assign the count to a variable before looping. Otherwise, we'll keep asking REAPER for the count for every iteration of the loop, which might be expensive with a lot of events. I do see I made the same error in my original code. :)